fastscanner/services/indicators/service.py

```diff
@@ -136,18 +157,95 @@ async def handle(self, channel_id: str, data: dict[Any, Any]) -> None:
             if "timestamp" not in data:
                 logger.warning(f"Missing timestamp in message from {channel_id}")
                 return
+
             ts = pd.to_datetime(int(data["timestamp"]), unit="ms", utc=True).tz_convert(
                 LOCAL_TIMEZONE_STR
             )
             new_row = pd.Series(data, name=ts)
 
-            for indicator in self._indicators:
-                new_row = await indicator.extend_realtime(self._symbol, new_row)
-
-            self._handler.handle(self._symbol, new_row)
+            await self._add_to_buffer(new_row)
 
         except Exception as e:
             logger.error(
                 f"[Handler Error] Failed processing message from {channel_id}: {e}"
             )
+
+    async def _add_to_buffer(self, new_row: pd.Series):
+        assert isinstance(new_row.name, pd.Timestamp)
+        ts: pd.Timestamp = new_row.name
+        candle_start = self._get_interval_start(ts)
+        logger.debug(
+            f"[{self._symbol}] Buffer updated at {new_row.name} | Buffer keys: {[k for k in self._buffer.keys()]}"
+        )
+
+        if (
+            not self._buffer
+            or self._get_interval_start(next(iter(self._buffer))) != candle_start
+        ):
```

Comment:
This conditional doesn't make sense. self._get_interval_start should always be the same as candle_start because before the next candle arrives, the buffer should be flushed. If they're different the bug is there.